hardware/opentrons_hardware/drivers/can_bus/settings.py

```
@@ -31,6 +50,48 @@ class DriverSettings(BaseSettings):
 
     host: str = Field(DEFAULT_HOST, description=f"{OPENTRONS_INTERFACE} only.")
     port: int = Field(DEFAULT_PORT, description=f"{OPENTRONS_INTERFACE} only.")
+    fcan_clock: int = Field(DEFAULT_FDCAN_CLK, description="pcan only.")
+    sample_rate: float = Field(DEFAULT_SAMPLE_RATE, description="pcan only.")
 
     class Config:  # noqa: D106
         env_prefix = "OT3_CAN_DRIVER_"
+
+
+def calculate_fdcan_parameters(
+    fcan_clock: Optional[int], bitrate: Optional[int], sample_rate: Optional[float]
+) -> PCANParameters:
+    """Calculate FDCAN Timing Parameters.
+
+    Args:
+        fcan_clock: The clock used by the analyzer
+        bitrate: The bitrate of data transfer
+        sample_rate: When to sample the data.
+
+
+    """
+    if not fcan_clock:
+        fcan_clock = DEFAULT_FDCAN_CLK
+    if not bitrate:
+        bitrate = DEFAULT_BITRATE
+    if not sample_rate:
+        sample_rate = DEFAULT_SAMPLE_RATE
+
+    sample_rate_percent = sample_rate / 100
+    bit_time = 1 / bitrate  # us units
+    time_quanta_ns = 1 / fcan_clock
+    time_quanta_unitless = bit_time / time_quanta_ns
+
+    # Given a sample rate, we can say that
+    # (jump_width_seg + TSEG1)/(jump_width_seg + TSEG1 + TSEG2) = sample_rate
+    # and also that (jump_width_seg + TSEG1 + TSEG2) = unitless_time_quanta (tq)
+    tseg_1 = (sample_rate_percent * time_quanta_unitless) - DEFAULT_JUMP_WIDTH_SEG
+    tseg_2 = time_quanta_unitless - DEFAULT_JUMP_WIDTH_SEG - tseg_1
+    brp = fcan_clock // MINIMUM_FDCAN_CLK  # clock prescaler (unitless)
```

no, this would generate the clock divider to generate the minimum clock speed from the current clock speed. We need to generate the time quantum frequency from whatever specified clock it is - this should be `fcan_clk_mhz / (1/time_quantum_us)` and it should raise an exception if it doesn't end up being an integer